pkg/backend/httpstate/backend.go

```diff
@@ -646,9 +647,11 @@ func (b *cloudBackend) ParseStackReference(s string) (backend.StackReference, er
 		qualifiedName.Project = b.currentProject.Name.String()
 	}
 
-	if !tokens.IsName(qualifiedName.Name) {
-		return nil, errors.New("stack names may only contain alphanumeric, hyphens, underscores, and periods")
+	if err := validation.ValidateStackName(qualifiedName.Name); err != nil {
+		return nil, err
```

Comment:
This is technically a stricter validation, since `tokens.IsName` doesn't check the length. In practice, I believe that this will be benign because the service will validate the name as well and it should enforce the length limit.